src/main/java/org/diatliuk/bookstore/service/impl/UserServiceImpl.java

```diff
@@ -35,10 +40,22 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
         user.setPassword(passwordEncoder.encode(user.getPassword()));
         Role defaultRole = roleRepository
                 .findByName(DEFAULT_ROLE)
-                .orElseThrow(() -> new RoleNotFoundException("The role: "
+                .orElseThrow(() -> new EntityNotFoundException("The role: "
                                                             + DEFAULT_ROLE + " does not exist"));
         user.setRoles(Set.of(defaultRole));
         User savedUser = userRepository.save(user);
+        ShoppingCart shoppingCart = new ShoppingCart();
```

Comment:
I suggest to extract this logic to shopping cart related service and call service method here, instead of repository